### PR TITLE
Ze/auto add case types

### PR DESCRIPTION
**Pull request by zandre-eng**
## Technical Summary

Link to ticket [here](https://dimagi-dev.atlassian.net/browse/SC-2524?atlOrigin=eyJpIjoiZTg2Y2MxMDEzMjBiNGM5MThlMGFjMWFkYzJmNzNkMmMiLCJwIjoiaiJ9).

When the user creates a SaveToCase question and fills in a new case type, this will now be reflected in the data dictionary when the user clicks on the "Click to sync" button in the Data Dictionary page.

## Feature Flag

`SaveToCase`

### Safety story

Did testing on local environment and staging.

### Automated test coverage

No automated tests.

### QA Plan

No QA.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by sravfeyn — APPROVED** [review]

File: corehq/apps/app_manager/dbaccessors.py [hunk 1]
```diff
@@ -500,16 +501,24 @@ def get_version_build_id(domain, app_id, version):
 def get_case_types_from_apps(domain):
     """
     Get the case types of modules in applications in the domain.
+    Also returns case types for SaveToCase properties in the domain, if the toggle is enabled.
     :returns: A set of case_types
     """
```
Comment (SmittieC) [inline corehq/apps/app_manager/dbaccessors.py:506]: I can't comment on the logic itself, but I'll offer a nitpick on the code:

To make it easier to read, I'd split out this change into its own function. Something like
```python
save_to_case_case_types = _get_save_to_case_case_types(domain) # ugh, naming is hard
```

File: corehq/apps/app_manager/dbaccessors.py [hunk 2]
```diff
@@ -500,16 +501,24 @@ def get_version_build_id(domain, app_id, version):
...
+
     case_types_agg = NestedAggregation('modules', 'modules').aggregation(
         TermsAggregation('case_types', 'modules.case_type.exact'))
     q = (AppES()
          .domain(domain)
          .is_build(False)
          .size(0)
          .aggregation(case_types_agg))
-    return set(q.run().aggregations.modules.case_types.keys) - {''}
+    return set(q.run().aggregations.modules.case_types.keys).union(save_to_case_updates) - {''}
```
Comment (SmittieC) [inline corehq/apps/app_manager/dbaccessors.py:521]: Then I'd change this line into two lines:
```python
case_types = set(q.run().aggregations.modules.case_types.keys)
return save_to_case_case_types.union(case_types) - {''}
```

Comment (zandre-eng) [reply corehq/apps/app_manager/dbaccessors.py]: Some good points, the code readability could definitely be improved. I have done some changes as per your suggestions. Thanks for pointing them out :+1:

**Review by SmittieC — APPROVED** [review]
Thanks for the refactoring. LGTM :rocket

**Pull request merged**